utensor_cgen/backend/utensor/_graph_lower/_op_lower.py

```diff
@@ -25,11 +31,10 @@ def default_config(cls):
     return {}
 
 
-class uTensorRearchGraphLower(BackendPart):
-  TARGET = 'utensor'
+class uTensorRearchGraphLower(uTensorGraphLowerBase):
   PART = 'rearch_graph_lower'
 
-  class OptypRenameManager(object):
+  class OptypeRenameManager(object):
```

Comment:
Something that might be useful is a specification for the ops on the backend that describe names, inputs, outputs, options (passed to the constructor) and potentially generic knowledge of sizes/validation functions. We kind of already have this so it might be easy to link in here vs the select op stage later.

Reply:
According to my understanding, you are saying that we shift the checks from the code generator to the lowering phase right?
It's doable but I think it need some refactoring work to do.
I'll put it in the issue tracker. How do you think?

Reply:
We should probably discuss this offline first, there is a lot in this tiny post :)